File: src/types/excel.ts
export interface ExcelFileInfo {
  path: string;
  name: string;
  size: number;
}

export interface SheetInfo {
  name: string;
  rowCount: number;
  columnCount: number;
}

export interface ColumnInfo {
  name: string;
  sampleValues: string[];
}

export interface PreviewData {
  columns: string[];
  rows: Record<string, string>[];
  totalRows: number;
}

export interface FileSelection {
  file: ExcelFileInfo;
  fileIndex: number;
  sheetInfo: SheetInfo[];
  selectedSheets: string[];
  columnInfo: Record<string, ColumnInfo[]>;
  selectedColumns: Record<string, string[]>;
  previewData: Record<string, PreviewData>;
}

export interface SampleData {
  columns: ColumnInfo[];
  rows: string[][];
  totalRows: number;
  sampleSize: number;
}

export interface ColumnData {
  columns: string[];
  rows: string[][];
  totalRows: number;
}

export interface WriteResult {
  row: number;
  value: string;
}

export interface ApplyFormulaRequest {
  path: string;
  sheet: string;
  column: string;
  formula: string;
}

export interface ProcessingStatus {
  totalRows: number;
  processedRows: number[];
  resultColumn: string;
}

File: src-tauri/src/models/excel.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct ExcelInfo {
    pub file_path: String,
    pub file_name: String,
    pub file_size: u64,
    pub sheets: Vec<SheetMeta>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct SheetMeta {
    pub name: String,
    pub row_count: usize,
    pub column_count: usize,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct ColumnInfo {
    pub name: String,
    pub index: usize,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct ColumnData {
    pub columns: Vec<String>,
    pub rows: Vec<Vec<String>>,
    pub total_rows: usize,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct SampleData {
    pub columns: Vec<ColumnInfo>,
    pub rows: Vec<Vec<String>>,
    pub total_rows: usize,
    pub sample_size: usize,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct WriteResult {
    pub row: usize,
    pub value: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct WriteResultsRequest {
    pub path: String,
    pub sheet: String,
    pub column: String,
    pub results: Vec<WriteResult>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct ApplyFormulaRequest {
    pub path: String,
    pub sheet: String,
    pub column: String,
    pub formula: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct ProcessingStatus {
    pub total_rows: usize,
    pub processed_rows: Vec<usize>,
    pub result_column: String,
}
